cal/cal/static/js/app.js

```diff
@@ -115,55 +115,54 @@ analyticsApp.component('tagList', {
   }
 });
 
-function TagsDetailCtrl($scope, $http) {
-  var tagUrl = '/v1/tags/' + this.tagId + '/events';
-  var eventweek = '/v1/tags/' + this.tagId + '/eventWeek';
-  var query_timezone = moment.tz.guess();
-  $scope.tagDetails = [];
-  $scope.tagEvents = [];
-  $scope.tagHours = this.tagHours;
-
-  $http({method: 'GET', url: tagUrl + '.json' }).
-  success(function successCallback(data) {
-    for (var i = 0; i < data.results.length; i++) {
-      var event = data.results[i];
-      $scope.tagEvents.push({
-        start: (new Date(event.start)).toString(),
-        name: event.name,
-      });
-    }
-  });
-  $http({
-    method: 'GET',
-    url: eventweek + '.json',
-    params: {
-      timezone: query_timezone,
-    }
-  }).
-  success(function successCallback(data) {
-    $scope.tagHours = $scope.tagHours / data.length;
+function TimelineBaseCtrl($scope, $http) {
+  $scope.basePopulateData = function(data, type) {
+    var maxYValue = 0;
     var events = [];
-    var max_hour = 0; // Used to calculate max hour across events. Used in line graph for max y-axis.
     for (var i = 0; i < data.length; i++) {
       var event = data[i];
-      var start = new Date(event[0]);
-      var hour = event[1];
-      if (hour > max_hour) {
-        max_hour = hour;
-      }
+      var date = new Date(event[0]);
+      var hours = event[1];
+      if (hours > maxYValue) {
+        maxYValue = hours;
+      };
       events.push({
-        x: start,
-        y: hour
+        x: date,
+        y: hours
       });
-    }
-    $scope.tagDetails.push({
-      values: events,      //values - represents the array of {x,y} data points
-      key: 'Tag Graph', //key  - the name of the series.
-      color: '#003057',  //color - optional: choose your own line color.
+    };
+    $scope.ctrlDetails = [];
```

Comment:
It's better style so if we want to add multiple lines we can just push more lines. You can change it for this purpose cuz we'll only have 1 line on detail pages.